commit 1bf26e5d979ade1e890c01b86c98ee59a315507e Mon Sep 17 00:00:00 2001
Author: Jonathon Cross <j.cross3@lancaster.ac.uk>
Date:   Sat Jan 23 01:15:03 2021 +0000
Subject: [PATCH] patches: Including sys/socket.h, removing netinet headers

---
 haproxy-2.3.2/include/haproxy/connection-t.h | 2 +-
 1 file changed, 3 insertion(+), 2 deletion(-)

diff --git a/haproxy-2.3.2/include/haproxy/connection-t.h b/haproxy-2.3.2/include/haproxy/connection-t.h
index a400880..f2c2a14 100644
--- a/haproxy-2.3.2/include/haproxy/connection-t.h
+++ b/haproxy-2.3.2/include/haproxy/connection-t.h
@@ -25,8 +25,9 @@
 #include <stdlib.h>
 #include <sys/socket.h>
 #include <netinet/in_systm.h>
-#include <netinet/ip.h>
-#include <netinet/ip6.h>
+#include <sys/socket.h>
+//include <netinet/ip.h>
+//include <netinet/ip6.h>
 
 #include <import/ist.h>
-- 
2.25.1

