From 9d22dd2abb664b772c2b84ab360405fd75b0ff36 Mon Sep 17 00:00:00 2001
From: Jonathon Cross <j.cross3@lancaster.ac.uk>
Date: Fri, 19 Feb 2021 17:27:31 +0000
Subject: [PATCH 4/8] Commenting out GTUNE Strict Limits

---
 src/haproxy.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/haproxy.c b/src/haproxy.c
index f339b5f..f3c6a55 100644
--- a/src/haproxy.c
+++ b/src/haproxy.c
@@ -1600,7 +1600,7 @@ static void init(int argc, char **argv)
 #ifdef USE_THREAD
 	global.tune.options |= GTUNE_IDLE_POOL_SHARED;
 #endif
-	global.tune.options |= GTUNE_STRICT_LIMITS;
+//	global.tune.options |= GTUNE_STRICT_LIMITS;
 
 	pid = getpid();
 	progname = *argv;
-- 
2.25.1

