commit b740f74528d96d1f5f2ea08e17d683589eda925a Mon Sep 17 00:00:00 2001
Author: Jonathon Cross <j.cross3@lancaster.ac.uk>
Date:   Sun Jan 31 15:08:41 2021 +0000
Subject: [PATCH] patches: Removing irrelevant group identity code
---
 haproxy-2.3.2/src/mworker-prog.c | 6 +-
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/haproxy-2.3.2/src/mworker-prog.c b/haproxy-2.3.2/src/mworker-prog.c
index e77317f..a8e1b72 100644
--- a/haproxy-2.3.2/src/mworker-prog.c
+++ b/haproxy-2.3.2/src/mworker-prog.c
@@ -92,8 +92,8 @@ int mworker_ext_launch_all()
                                mworker_unblock_signals();
                                mworker_cleanlisteners();
                                mworker_cleantasks();
-
-                               /* setgid / setuid */
+
+                               /* setgid / setuid
                                if (child->gid != -1) {
                                        if (getgroups(0, NULL) > 0 && setgroups(0, NULL) == -1)
                                                ha_warning("[%s.main()] Failed to drop supplementary groups. Using 'gid'/'group'"
@@ -109,7 +109,7 @@ int mworker_ext_launch_all()
                                        ha_alert("[%s.main()] Cannot set uid %d.\n", child->command[0], child->gid);
                                        exit(1);
                                }
-
+                               */
                                execvp(child->command[0], child->command);

                                ha_alert("Cannot execute %s: %s\n", child->command[0], strerror(errno));
-- 
2.25.1

