commit b6fa156485a7ecaab58270deae95c881204ae2c4 Mon Sep 17 00:00:00 2001
Author: Jonathon Cross <j.cross3@lancaster.ac.uk>
Date:   Sun Jan 31 15:35:37 2021 +0000
Subject: [PATCH] patches: Changing memory allocation/freeing from mmap/munmap

---
 haproxy-2.3.2/src/shctx.c | 10 +-
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/haproxy-2.3.2/src/shctx.c b/haproxy-2.3.2/src/shctx.c
index f4002a4..2813974 100644
--- a/haproxy-2.3.2/src/shctx.c
+++ b/haproxy-2.3.2/src/shctx.c
@@ -315,9 +315,9 @@ int shctx_init(struct shared_context **orig_shctx, int maxblocks, int blocksize,
        if (shared)
                maptype = MAP_SHARED;
 #endif
+
+       shctx = (struct shared_context *)malloc(sizeof(struct shared_context) + extra + (maxblocks * (sizeof(struct shared_block) + blocksize)));

-       shctx = (struct shared_context *)mmap(NULL, sizeof(struct shared_context) + extra + (maxblocks * (sizeof(struct shared_block) + blocksize)),
-                                             PROT_READ | PROT_WRITE, maptype | MAP_ANON, -1, 0);
        if (!shctx || shctx == MAP_FAILED) {
                shctx = NULL;
                ret = SHCTX_E_ALLOC_CACHE;
@@ -330,7 +330,7 @@ int shctx_init(struct shared_context **orig_shctx, int maxblocks, int blocksize,
        if (maptype == MAP_SHARED) {
 #ifdef USE_PTHREAD_PSHARED
                if (pthread_mutexattr_init(&attr)) {
-                       munmap(shctx, sizeof(struct shared_context) + extra + (maxblocks * (sizeof(struct shared_block) + blocksize)));
+                       free(shctx);
                        shctx = NULL;
                        ret = SHCTX_E_INIT_LOCK;
                        goto err;
@@ -338,7 +338,7 @@ int shctx_init(struct shared_context **orig_shctx, int maxblocks, int blocksize,

                if (pthread_mutexattr_setpshared(&attr, PTHREAD_PROCESS_SHARED)) {
                        pthread_mutexattr_destroy(&attr);
-                       munmap(shctx, sizeof(struct shared_context) + extra + (maxblocks * (sizeof(struct shared_block) + blocksize)));
+                       free(shctx);
                        shctx = NULL;
                        ret = SHCTX_E_INIT_LOCK;
                        goto err;
@@ -346,7 +346,7 @@ int shctx_init(struct shared_context **orig_shctx, int maxblocks, int blocksize,

                if (pthread_mutex_init(&shctx->mutex, &attr)) {
                        pthread_mutexattr_destroy(&attr);
-                       munmap(shctx, sizeof(struct shared_context) + extra + (maxblocks * (sizeof(struct shared_block) + blocksize)));
+                       free(shctx);
                        shctx = NULL;
                        ret = SHCTX_E_INIT_LOCK;
                        goto err;
-- 
2.25.1

