commit 824f3eb839bd6399843a35d472b6a53e9a9efcaa Mon Sep 17 00:00:00 2001
Author: Jonathon Cross <j.cross3@lancaster.ac.uk>
Date:   Sun Jan 31 15:37:49 2021 +0000
Subject: [PATCH] patches: Commenting out proprietary code

---
 haproxy-2.3.2/src/xprt_handshake.c | 2 +-
 1 file changed, 2 insertion(+)

diff --git a/haproxy-2.3.2/src/xprt_handshake.c b/haproxy-2.3.2/src/xprt_handshake.c
index 752a4f3..08e8f69 100644
--- a/haproxy-2.3.2/src/xprt_handshake.c
+++ b/haproxy-2.3.2/src/xprt_handshake.c
@@ -57,11 +57,13 @@ static struct task *xprt_handshake_io_cb(struct task *t, void *bctx, unsigned sh
                }

        if (conn->flags & CO_FL_ACCEPT_CIP)
+               #if 0
                if (!conn_recv_netscaler_cip(conn, CO_FL_ACCEPT_CIP)) {
                        ctx->xprt->subscribe(conn, ctx->xprt_ctx, SUB_RETRY_RECV,
                            &ctx->wait_event);
                        goto out;
                }
+               #endif /* 0 */

        if (conn->flags & CO_FL_ACCEPT_PROXY)
                if (!conn_recv_proxy(conn, CO_FL_ACCEPT_PROXY)) {
-- 
2.25.1

